modules/repository/generate.go

```diff
@@ -97,6 +102,66 @@ func checkGiteaTemplate(tmpDir string) (*models.GiteaTemplate, error) {
 	return gt, nil
 }
 
+func getSubmodules(repoDir string) []submodule {
```

Comment:
move this into the `modules/git` package

Reply:
Moved it and added a basic unittest. The extra test repository clutters the PR up a good bit, is there a better way to add those?

Reply:
no this is just fine :+1: